pool/src/pool/status.rs

```diff
@@ -1,59 +1,93 @@
-    if pool_backstop_data.q4w_pct >= 0_5000000 {
-        pool_config.status = 2;
-    } else if pool_backstop_data.q4w_pct >= 0_2500000
-        || calc_pool_backstop_threshold(&pool_backstop_data) < SCALAR_7
-    {
-        pool_config.status = 1;
+    let threshold = calc_pool_backstop_threshold(&pool_backstop_data);
+    let mut met_threshold = true;
+    if threshold < SCALAR_7 {
+        met_threshold = false;
+    }
+    // even numbered statuses are admin controlled
+    if pool_status % 2 == 0 {
+        let admin = storage::get_admin(&e);
+        // admin auth required to set admin status'
+        admin.require_auth();
+        match pool_status {
+            0 => {
+                // Threshold must be met and q4w must be under 50% for the admin to set Active
+                if !met_threshold || pool_backstop_data.q4w_pct >= 0_5000000 {
+                    panic_with_error!(e, PoolError::BadRequest);
+                }
+                // Admin Active
+                pool_config.status = 0;
+            }
+            2 => {
+                // Q4w must be under 75% for admin to set On-Ice
+                if pool_backstop_data.q4w_pct >= 0_7500000 {
+                    panic_with_error!(e, PoolError::BadRequest);
+                }
+                // Admin On-Ice
+                pool_config.status = 2;
+            }
+            4 => {
+                // Admin can always freeze the pool
+                // Admin Frozen
+                pool_config.status = 4;
+            }
+            _ => {
+                panic_with_error!(e, PoolError::BadRequest);
+            }
+        }
+        e.events()
+            .publish((Symbol::new(&e, "set_status"), admin), pool_status);
     } else {
-        pool_config.status = 0;
+        // Admin Frozen supersedes all other statuses
+        if pool_config.status == 4 {
+            panic_with_error!(e, PoolError::BadRequest);
+        // Q4W over 75% freezes the pool
+        // Q4W over 60% freezes the pool if it is not Admin On-Ice
+        } else if pool_backstop_data.q4w_pct >= 0_7500000
+            || (pool_backstop_data.q4w_pct >= 0_6000000 && pool_config.status != 2)
+        {
+            // Frozen
+            pool_config.status = 5;
+        // Admin On-Ice supersedes backstop triggered on-ice
+        // Q4w over 50% puts the pool on-ice
+        // Threshold not being met puts the pool on-ice
+        // As long as pool isn't admin active q4w over 30% puts pool on-ice
+        } else if pool_config.status != 2
+            && (pool_backstop_data.q4w_pct >= 0_5000000
+                || !met_threshold
+                || (pool_config.status != 0 && pool_backstop_data.q4w_pct >= 0_3000000))
+        {
+            // On-Ice
+            pool_config.status = 3;
+        // Admin status' all supersede backstop triggered active
+        } else if pool_config.status % 2 != 0 {
+            // Active
+            pool_config.status = 1;
+        }
```

Comment:
would it be easier to frame this as a switch? I have a hard time following the if statements.

```rust
match cur_status {
    4 -> panic
    2 -> check if we can set to 5
    0 -> check if we can set to 3
    else ->    if pool_backstop_data.q4w_pct >= 0_5000000 {
        pool_config.status = 5;
    } else if pool_backstop_data.q4w_pct >= 0_2500000
        || calc_pool_backstop_threshold(&pool_backstop_data) < SCALAR_7
    {
        pool_config.status = 3;
    } else {
        pool_config.status = 1;
}
```

at the very least, it might be worth trying to make the code a bit more follow-able. If it doesn't, I am cool with this.

Reply:
match statement helped